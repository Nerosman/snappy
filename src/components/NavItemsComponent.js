import React from "react";
import {Link} from "react-router-dom";
import {NavItem} from "reactstrap";

export const NavItemsComponent = () => {
    switch (window.location.pathname) {
        case "/ship-info":
            return <NavItem>
                <Link className="nav-link" to="/admin-panel">Admin panel</Link>
            </NavItem>;
        case "/admin-panel":
            return <NavItem>
                <Link className="nav-link" to="/ship-info">Get gift</Link>
            </NavItem>;
        default :
            return <React.Fragment>
                <NavItem>
                    <Link className="nav-link" to="/admin-panel">Admin panel</Link>
                </NavItem>
                <NavItem>
                    <Link className="nav-link" to="/ship-info">Get gift</Link>
                </NavItem>
            </React.Fragment>
    }
};
