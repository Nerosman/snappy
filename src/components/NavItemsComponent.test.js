import {configure, shallow} from "enzyme/build";
import {NavItemsComponent} from "./NavItemsComponent";
import React from "react";
import Adapter from "enzyme-adapter-react-16/build";

configure({ adapter: new Adapter() });

describe('<NavItemsComponent />', () => {
    test('take snapshot for navItemsComponent', () => {
        const wrapper = shallow(
            <NavItemsComponent />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should work with a switch statement with /ship-info pathname', () => {
        const wrapper = shallow(<NavItemsComponent pathname={"/ship-info"} />);
        expect(wrapper);
    });

    it('should work with a switch statement with /admin-panel pathname', () => {
        const wrapper = shallow(<NavItemsComponent pathname={"/admin-panel"} />);
        expect(wrapper);
    });
});
