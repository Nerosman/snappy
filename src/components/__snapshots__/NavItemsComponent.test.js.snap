// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavItemsComponent /> take snapshot for navItemsComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavItemsComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <NavItem
          tag="li"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/admin-panel"
          >
            Admin panel
          </Link>
        </NavItem>,
        <NavItem
          tag="li"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/ship-info"
          >
            Get gift
          </Link>
        </NavItem>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Link
            className="nav-link"
            replace={false}
            to="/admin-panel"
          >
            Admin panel
          </Link>,
          "tag": "li",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Admin panel",
            "className": "nav-link",
            "replace": false,
            "to": "/admin-panel",
          },
          "ref": null,
          "rendered": "Admin panel",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Link
            className="nav-link"
            replace={false}
            to="/ship-info"
          >
            Get gift
          </Link>,
          "tag": "li",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Get gift",
            "className": "nav-link",
            "replace": false,
            "to": "/ship-info",
          },
          "ref": null,
          "rendered": "Get gift",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavItem
            tag="li"
          >
            <Link
              className="nav-link"
              replace={false}
              to="/admin-panel"
            >
              Admin panel
            </Link>
          </NavItem>,
          <NavItem
            tag="li"
          >
            <Link
              className="nav-link"
              replace={false}
              to="/ship-info"
            >
              Get gift
            </Link>
          </NavItem>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Link
              className="nav-link"
              replace={false}
              to="/admin-panel"
            >
              Admin panel
            </Link>,
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Admin panel",
              "className": "nav-link",
              "replace": false,
              "to": "/admin-panel",
            },
            "ref": null,
            "rendered": "Admin panel",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Link
              className="nav-link"
              replace={false}
              to="/ship-info"
            >
              Get gift
            </Link>,
            "tag": "li",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Get gift",
              "className": "nav-link",
              "replace": false,
              "to": "/ship-info",
            },
            "ref": null,
            "rendered": "Get gift",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
